pkg/agent/svid/rotator.go

```diff
@@ -137,6 +138,29 @@ func (r *rotator) SetRotationFinishedHook(f func()) {
 	r.rotationFinishedHook = f
 }
 
+func (r *rotator) Reattest(ctx context.Context) (err error) {
+	state, ok := r.state.Value().(State)
+	if !ok {
+		return fmt.Errorf("unexpected value type: %T", r.state.Value())
+	}
+
+	if state.Reattestable {
+		if !r.c.DisableReattestToRenew {
+			err = r.reattest(ctx)
+		} else {
+			return errors.New("re-attestation is disabled")
```

Comment:
That's what i had before and @amartinezfayo asked me to change. Either way is fine with me.

`rotateSVIDIfNeeded` can't be used directly because that only rotates if the SVID is expired.